### PR TITLE
fix(form): native submit event should not trigger an error

### PR DESCRIPTION
**Pull request by daKmoR**
closes https://github.com/ing-bank/lion/issues/42

### CODE REVIEW

File: packages/form/src/LionForm.js [hunk 1]
```diff
@@ -18,16 +18,23 @@ export class LionForm extends DelegateMixin(LionFieldset) {
...
+    this.addEventListener('submit', this.__boundSubmit);
+    this.addEventListener('reset', this.__boundReset);
   }
 
   disconnectedCallback() {
     super.disconnectedCallback();
-    this.removeEventListener('submit', this._submit);
-    this.removeEventListener('reset', this._reset);
+    this.removeEventListener('submit', this.__boundSubmit);
+    this.removeEventListener('reset', this.__boundReset);
```
Comment (bashmish) [inline packages/form/src/LionForm.js]: let's be more consistent :)

image

File: packages/form/test/lion-form.test.js [hunk 2]
```diff
@@ -37,4 +37,22 @@ describe('<lion-form>', () => {
       firstName: 'Foo',
     });
   });
+
+  it('works with the native submit event (triggered via a button)', async () => {
+    let isSubmitted = false;
+    const mySubmit = () => {
+      isSubmitted = true;
+    };
```
Comment (bashmish) [inline packages/form/test/lion-form.test.js]: maybe use a spy here and `spy.calledOnce` in the end to make test even more readable and prevent double firing of the event on the lion-form (which I expect to happen if not properly handled)

Comment (bashmish) [reply packages/form/test/lion-form.test.js]: I would also double check other tests and if they test this with `calledOnce`, I'm really afraid this MR can cause double submit errors

File: packages/form/src/LionForm.js (hunk 1 above)
Comment (daKmoR) [reply packages/form/src/LionForm.js]: empty line removed

File: packages/form/test/lion-form.test.js (hunk 2 above)
Comment (daKmoR) [reply packages/form/test/lion-form.test.js]: used sinon now

**Pull request merged**